controllers/idler/idler_controller.go

```diff
@@ -141,13 +141,18 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
 				}
 
 				// Check if it belongs to a controller (Deployment, DeploymentConfig, etc) and scale it down to zero.
-				appType, appName, deletedByController, err := r.scaleControllerToZero(podLogger, pod.ObjectMeta)
+				appType, appName, deletedByController, err := func() (string, string, bool, error) {
+					lctx, lcancel := context.WithCancel(ctx)
+					defer lcancel()
```

Comment:
why is this needed?

Reply:
that's an error due to an improper use of `log.IntoContext`. let me refactor this